### PR TITLE
[frontend] Improve error messages for web app.

### PR DESCRIPTION
**Pull request by fwpushan**

### CODE REVIEW

File: frontend/src/hooks/useError.ts [hunk 1]
```diff
@@ -1,21 +1,51 @@
...
+  prefix?: string;
 }) {
-  const { updateErrorContext } = useContext(ErrorContext)
+  const { updateErrorContext } = useContext(ErrorContext);
   useEffect(() => {
     if (error) {
-      error.message = error.message || 'An unknown error occurred.'
-      if (error.response?.data?.message) {
-        error.message = error.response?.data?.message
+      let message = error.message || 'An unknown error occurred.';
```
Comment (will-freshworks) [inline frontend/src/hooks/useError.ts:15]: Do we want undefined error to not show, or do we want `'An unknown error occurred.'`? Might be worth using `let message = error?.message || ...` and dropping the `if (error) { ... }`

File: frontend/src/hooks/useError.ts [hunk 2]
```diff
@@ -1,21 +1,51 @@
...
+            }
+          }
+        }
+
+        const errorMessage = `${message} (${error.response?.statusText})`;
+        const finalMessage = prefix
+          ? `${prefix} Error: ${errorMessage}`
+          : `Unable to perform operation. Error: ${errorMessage}`;
+        updateErrorContext(finalMessage);
+      } catch (excp) {
```
Comment (will-freshworks) [inline frontend/src/hooks/useError.ts:44]: Not sure I like a try/catch around pure object manipulation code. I would assume any error is a missing case in the if/elses.

File: frontend/src/hooks/useError.ts [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
+        const finalMessage = prefix ? `${prefix} ${message}` : message;
+        updateErrorContext(finalMessage);
       }
-      const message = prefix ? `${prefix} ${error.message}` : error.message
-      updateErrorContext(message)
     }
-  }, [error, prefix, updateErrorContext])
-  return
+  }, [error, prefix, updateErrorContext]);
+  return;
```
Comment (will-freshworks) [inline frontend/src/hooks/useError.ts:50]: redundant return

File: frontend/src/hooks/useError.ts [hunk 4]
```diff
@@ -1,21 +1,51 @@
...
-import { AxiosError } from 'axios'
-import ErrorContext from 'contexts/ErrorContext'
+import { useEffect, useContext } from 'react';
+import { AxiosError } from 'axios';
+import ErrorContext from 'contexts/ErrorContext';
 
-export default function useError ({ error, prefix }: {
-  error: AxiosError | undefined,
-  prefix?: string,
+export default function useError({
```
Comment (will-freshworks) [inline frontend/src/hooks/useError.ts:5]: Love this custom hook!

**Review by will-freshworks — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useError.ts (hunk 2 above)
Comment (fwpushan) [reply frontend/src/hooks/useError.ts:44]: @will-freshworks , I put this because in case axios error response structure changes .....

**Review by jerryfreshworks — APPROVED** [review]

**Pull request merged**